### ISSUE TITLE
[Feature Request] Adding Eagle, Medusa, Look Ahead decoding ( improvements of Speculative decoding)

### ISSUE THREAD

**Issue by HamidShojanazeri** [vllm-project/vllm#2791]
Thanks for the great work team. I wonder if there is any plan to add new improvements to speculative decoding such as [Eagle](https://sites.google.com/view/eagle-llm), [Medusa](https://sites.google.com/view/medusa-llm), [look ahead decoding](https://github.com/hao-ai-lab/LookaheadDecoding). These could result in accumulative speed ups for VLLM.

**Comment by simon-mo** [comment]
Yes. The plan is here #1797

**Comment by HamidShojanazeri** [comment]
thanks for sharing @simon-mo that sounds great! I also wonder if newer methods also can improve on speculative decoding with removing the need for a draft model and we are exploring that path as well?

**Comment by simon-mo** [comment]
The speculative decoding framework is designed to support a wide range of draft model and draft model free algorithms. Once the immediate features are in place (by @cadedaniel), we welcome community's contribution for more methods!

**Comment by cadedaniel** [comment]
Correct! And yes, speculation methods without a draft model have benefits in both performance and usability. Unclear right now which specific approach will end up being the best but vLLM should support it.

**Comment by caliber1313** [comment]
I would like to suggest Hydra in your project alongside with medusa. Please find hydra repository here: https://github.com/zankner/Hydra. 

Thank you for your consideration

**Comment by mgoin** [comment]
This has been implemented!
